owncloudApp/src/main/res/layout/passcodelock.xml

```diff
@@ -63,26 +68,55 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:textColor="@color/warning"
-        android:gravity="center_horizontal"
         android:textSize="16sp"
-        />
-
-    <androidx.appcompat.widget.AppCompatButton
-        android:id="@+id/cancel"
-        android:theme="@style/Button.Primary"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="@android:string/cancel" />
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/layout_code" />
 
     <TextView
         android:id="@+id/lock_time"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginTop="@dimen/activity_vertical_margin"
+        android:layout_marginTop="@dimen/passcode_standard_margin"
         android:textColor="@android:color/black"
-        android:gravity="center_horizontal"
         android:textSize="16sp"
         android:visibility="invisible"
-        />
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/error" />
+
+    <com.owncloud.android.presentation.ui.security.passcode.NumberKeyboard
+        android:id="@+id/numberKeyboard"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginTop="@dimen/passcode_margin_top"
+        android:layout_marginBottom="@dimen/passcode_margin_bottom"
+        app:layout_constraintTop_toBottomOf="@id/lock_time"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintBottom_toTopOf="@id/btnCancel" />
+
+    <Button
+        android:id="@+id/btnCancel"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="@dimen/passcode_cancelBtn_margin_bottom"
+        android:paddingHorizontal="32dp"
+        android:text="@android:string/cancel"
+        android:textColor="@color/white"
+        android:background="@color/color_accent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
 
-</LinearLayout>
+    <!--
+    <Button
+        android:id="@+id/btnCancel"
+        android:theme="@style/Button.Primary"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@android:string/cancel"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"/>
+        -->
```

Comment:
Try to keep the code as clean as possible, if the button is commented for any reason, I encourage you to add a comment about it. If it is not used, remove it.